lib/oli/delivery/sections.ex

```diff
@@ -1028,6 +1034,36 @@ defmodule Oli.Delivery.Sections do
     end)
   end
 
+  # Returns a mapset of two element tuples of the form {source_resource_id, target_resource_id}
+  # representing the relates_to relationship between pages.
+  defp get_relates_to(publication_ids) do
+    joined_publication_ids = Enum.join(publication_ids, ",")
+    page_type_id = Oli.Resources.ResourceType.get_id_by_type("page")
+
+    sql = """
+    select
+      rev.resource_id, rev.relates_to
+    from published_resources as mapping
+    join revisions as rev
+    on mapping.revision_id = rev.id
+    where rev.resource_type_id = #{page_type_id} and array_length(rev.relates_to, 1) > 0 and mapping.publication_id IN (#{joined_publication_ids})
+    """
+
+    {:ok, %{rows: results}} = Ecto.Adapters.SQL.query(Oli.Repo, sql, [])
+
+    Enum.reduce(results, MapSet.new(), fn [source_id, relates_to], links ->
+
+      # The relates_to field is an array of resource ids, to be future proof
+      # to how relates_to is used, we will follow these 'links' in both directions
+      links = Enum.reduce(relates_to, links, fn target_id, links ->
+        MapSet.put(links, {source_id, target_id})
+      end)
+      Enum.reduce(relates_to, links, fn target_id, links ->
+        MapSet.put(links, {target_id, source_id})
+      end)
```

Comment:
Instead of going through the `relates_to` list twice, WDYT about doing this?
```suggestion
      Enum.reduce(relates_to, links, fn target_id, links ->
        MapSet.put(links, {source_id, target_id}) |> MapSet.put({target_id, source_id})
      end)
```